### PR TITLE
Fix bug that push data mode cannot be called by cpp

### PR DESCRIPTION
**Pull request by JackLau1222**
* fix PushData mode name in cpp interface
* implement Graph::Start(bool, bool) in cpp
* add Grpah::Close() and RealGraph::Close() function in cpp

### CODE REVIEW

File: bmf/engine/connector/include/builder.hpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ enum GraphMode {
     ServerMode,
     GeneratorMode,
     SubGraphMode,
-    UpdateMode,
```
Comment (sfeiwong) [inline bmf/engine/connector/include/builder.hpp:56]: Why "UpdateMode" was removed?

Comment (JackLau1222) [reply bmf/engine/connector/include/builder.hpp:56]: As i know, the graph mode has generator, sync, server, preload, subgraph and pushdata refer to https://babitmf.github.io/docs/bmf/multiple_features/graph_mode. 
And the engine code didn't use the update mode, so i think it's just typo error of PushData?

File: bmf/engine/connector/src/builder.cpp [hunk 2]
```diff
@@ -376,8 +376,8 @@ nlohmann::json RealGraph::Dump() {
     case SubGraphMode:
         info["mode"] = "Subgraph";
         break;
-    case UpdateMode:
```
Comment (sfeiwong) [inline bmf/engine/connector/src/builder.cpp:379]: Ditto, the UpdateMode tag was removed?

Comment (JackLau1222) [reply bmf/engine/connector/src/builder.cpp:379]: yes, please review the latest comment https://github.com/BabitMF/bmf/pull/155#discussion_r1897755729

**Pull request merged**